src/control/lib/daos/pool.go

```diff
@@ -0,0 +1,246 @@
+		PoolLayoutVer    uint32               `json:"pool_layout_ver"`
+		UpgradeLayoutVer uint32               `json:"upgrade_layout_ver"`
+	}
+
+	// StorageTargetUsage represents DAOS target storage usage
+	StorageTargetUsage struct {
+		Total     uint64           `json:"total"`
+		Free      uint64           `json:"free"`
+		MediaType StorageMediaType `json:"media_type"`
+	}
+)
+
+func (srs *StorageUsageStats) calcImbalance(targCount uint32) uint32 {
+	spread := srs.Max - srs.Min
+	return uint32((float64(spread) / (float64(srs.Total) / float64(targCount))) * 100)
+}
+
+func (pi *PoolInfo) Usage() []*PoolTierUsage {
+	var tiers []*PoolTierUsage
+	for _, tier := range pi.TierStats {
+		tiers = append(tiers, &PoolTierUsage{
+			TierName:  strings.ToUpper(tier.MediaType.String()),
+			Size:      tier.Total,
+			Free:      tier.Free,
+			Imbalance: tier.calcImbalance(pi.ActiveTargets),
+		})
+	}
+	return tiers
+}
+
+// RebuildState returns a string representation of the pool rebuild state.
+func (pi *PoolInfo) RebuildState() string {
+	if pi.Rebuild == nil {
+		return "Unknown"
+	}
+	return pi.Rebuild.State.String()
+}
+
+// Name retrieves effective name for pool from either label or UUID.
+func (pi *PoolInfo) Name() string {
+	name := pi.Label
+	if name == "" {
+		// use short version of uuid if no label
+		name = strings.Split(pi.UUID.String(), "-")[0]
+	}
+	return name
+}
+
+// PoolServiceState is used to represent the state of the pool service
+type PoolServiceState uint
+
+const (
+	// PoolServiceStateCreating indicates that the pool service is being created
+	PoolServiceStateCreating PoolServiceState = iota
+	// PoolServiceStateReady indicates that the pool service is ready to be used
+	PoolServiceStateReady
+	// PoolServiceStateDestroying indicates that the pool service is being destroyed
+	PoolServiceStateDestroying
+	// PoolServiceStateDegraded indicates that the pool service is in a degraded state
+	PoolServiceStateDegraded
+	// PoolServiceStateUnknown indicates that the pool service state is unknown
+	PoolServiceStateUnknown
+)
+
+func (pss PoolServiceState) String() string {
+	if pss < PoolServiceStateCreating || pss > PoolServiceStateUnknown {
+		return fmt.Sprintf("Invalid pool service state %d", pss)
+	}
+	return [...]string{
+		mgmtpb.PoolServiceState_Creating.String(),
+		mgmtpb.PoolServiceState_Ready.String(),
+		mgmtpb.PoolServiceState_Destroying.String(),
+		mgmtpb.PoolServiceState_Degraded.String(),
+		mgmtpb.PoolServiceState_Unknown.String(),
+	}[pss]
+}
+
+func (pss PoolServiceState) MarshalJSON() ([]byte, error) {
+	return []byte(`"` + pss.String() + `"`), nil
+}
+
+func (pss *PoolServiceState) UnmarshalJSON(data []byte) error {
+	stateStr := strings.Trim(string(data), "\"")
+
+	state, ok := mgmtpb.PoolServiceState_value[stateStr]
+	if !ok {
+		// Try converting the string to an int32, to handle the
+		// conversion from protobuf message using convert.Types().
+		si, err := strconv.ParseInt(stateStr, 0, 32)
```

Comment:
nice, approve of using this method that we've used elsewhere, handles the multiple cases well